Adding job workflows to upgrade guide (#9019)
fishtown-analytics/docs.getdbt.com: website/docs/guides/fusion-upgrade-prepare.md

2. Add a commit message like "Upgrade dbt packages for Fusion compatibility".
3. Click **Commit and sync**.

## Update your jobs

Use the <Constant name="fusion" /> readiness panel to validate your jobs against the <Constant name="fusion_engine" /> and fix any issues you find.

### Review your jobs

The readiness panel shows how many jobs are ineligible for <Constant name="fusion" /> or have an unknown eligibility status:

1. Open the jobs list using either path:
   - From the main menu, go to **Orchestration** → **Jobs**.
   - From the readiness panel, click **Review jobs**.
   <Lightbox src="/img/fusion/review-jobs.png" width="60%" title="Shortcut to review your jobs from the readiness panel"/>

> 2. Find the <Constant name="fusion" /> eligibility column to the right of your jobs. Click **Review job** next to any job that is ineligible or has an unknown eligibility status.

Reviewer: ```suggestion
2. Find the <Constant name="fusion" /> eligibility icon to the right of your jobs. Click **Review job** for  any job that is ineligible or has an unknown eligibility status.
```

   <Lightbox src="/img/fusion/job-eligibility.png" width="60%" title="Take action on your jobs to make them Fusion eligible."/>

> 3. Click **Debug on Fusion** and choose one of the following:

Reviewer: The UI currently says "Debug on Studio" but I wonder if it should just say "Debug"

<img width="558" height="522" alt="Image" src="https://github.com/user-attachments/assets/eb6da77a-d4ee-4e48-8631-706eb54c9175" />

Author: Good call out! This will change, it's incorrect in product as is and will revert to what's in the screenshot

   - [Debug in Studio](#debug-in-studio)
   - [Debug in Studio with Copilot](#debug-in-studio-with-copilot)
   - [Run once on Fusion](#run-once-on-fusion)

> :::note

Reviewer: We migth want this note up right after step 2 because that's where you'll need it. I was like _why is that drop down not available for my job_!?!

Author: Makes sense to me!

If a job has not run in the last 7 days, you must run it once for the debugging options to be available.
:::

#### Debug in Studio

> In the <Constant name="cloud_ide" />, run <Constant name="fusion" /> in your development environment to review project warnings and errors:

Reviewer: ```suggestion
In the <Constant name="studio_ide" />, run <Constant name="fusion" /> in your development environment to review project warnings and errors:
```

> 1. Click **Debug in Studio**. The <Constant name="cloud_ide" /> opens and the **Problems** tab appears.

Reviewer: ```suggestion
1. Click **Debug in Studio**. The <Constant name="cloud_ide" /> opens and selects the **Problems** tab.
```

> 1. Click **Debug in Studio**. The <Constant name="cloud_ide" /> opens and the **Problems** tab appears.

Author: ```suggestion
1. Click **Debug in Studio**. The <Constant name="studio_ide" /> opens and selects the **Problems** tab.
```

Author: Missed one

2. dbt sets only your development environment to <Constant name="fusion" />.

<Lightbox src="/img/fusion/fusion-ide.png" width="60%" title="Running Fusion in development"/>

3. Review the warnings or errors in the **Problems** tab.
4. Fix the issues directly or run the [autofix tool](/docs/cloud/studio-ide/autofix-deprecations).
5. When the project runs with no warnings or errors, publish your changes.
6. After you merge the changes, wait for the job to run again or run it manually.

#### Debug in Studio with Copilot <Lifecycle status="beta" size="80%" />

> If you have access to Copilot, use it to fix warnings or errors in the <Constant name="cloud_ide" />:

Reviewer: ```suggestion
If you have access to Copilot, use it to fix warnings or errors in the <Constant name="studio_ide" />:
```

1. Click **Debug in Studio with Copilot**.

> 2. dbt redirects you to the <Constant name="cloud_ide" /> and sets your development environment to <Constant name="fusion" />.

Reviewer: ```suggestion
2. dbt redirects you to the <Constant name="studio_ide" /> and sets your development environment to <Constant name="fusion" />.
```

3. Copilot opens with this prompt: "I need help fixing Fusion compatibility issues in this project. Please investigate and resolve any deprecation warnings or incompatibilities. Please use the migrating-dbt-core-to-fusion skill to guide this."

   <Lightbox src="/img/fusion/migration-prompt.png" width="60%" title="The Fusion migration prompt in Copilot"/>

> 4. Approve Copilot's permission requests so it can run the commands it needs.

Reviewer: Before this, I also have to answer this question:
> Would you like me to start by running dbt debug to verify your credentials and connection work on Fusion, or skip that and go straight to dbt-autofix?

Author: Let me get to this in a follow up. I didn't see this step, so when @mirnawong1 documents further, we'll try to get possible follow ups and document those as well.

> 5. When the project runs with no warnings or errors, publish your changes.

Reviewer: Should we add a step before this that says something like, "Guide Copilot through a series of questions about how to proceed."

Author: Added!

> 5. When the project runs with no warnings or errors, publish your changes.

Reviewer: Should we say "Commit and publish?"

Author: Yes!
